uaclient/entitlements/repo.py

```diff
@@ -72,7 +74,7 @@ def _perform_enable(self, silent: bool = False) -> bool:
         @raises: UserFacingError on failure to install suggested packages
         """
         self.setup_apt_config(silent=silent)
-        if self.packages:
+        if self.packages is not None:
```

Comment:
@lucasmoura I should've left my `packages` comment down here.

Before this change, `_perform_enable` only calls `self.install_packages` if self.packages is a list with at least one item.

For the fips on containers use case, we want self.packages to be empty, but we still want to call `self.install_packages` so that `FIPSCommonEntitlement.install_packages` runs to install the condition_packages.

So the way I thought to do that was to distinguish between "No packages to install, don't even bother calling install_packages" (`None`) and the fips on containers scenario which is "No packages to install, but still call install_packages because there are conditional_packages" (`[]`).

I'm open to other ways of accomplishing this though

Reply:
Okay, I see the problem now. What do you think if we move that logic into `install_packages` ? We would check inside that function if we actually have any packages to install. In that way, we would need to modify the `packages` property of the FIPS classes.

Reply:
Ahhh thats a good idea! I'll try that out